### ISSUE TITLE
v1.23.14 on GKE changes causing parsing issues for crio.conf

### ISSUE THREAD

**Issue by joncbenderkh** [nestybox/sysbox#623]
v1.23.14-gke.1800 changes /etc/default/kubelet to include
`--pod-infra-container-image=gke.gcr.io/pause:3.6@sha256:10008c36b4611b44db1229451675d5d7d86c7ddf4ef00f883d806a01547203f6`

This appears to be causing /etc/crio/crio.conf to have an extra quote in the pause_image line:
```  
[crio.image]
    pause_image = "gke.gcr.io/pause:3.6@sha256:10008c36b4611b44db1229451675d5d7d86c7ddf4ef00f883d806a01547203f6""
```
On 1.23.13 the pause_image line is not present at all.

**Comment by ctalledo** [comment]
Good catch @joncbenderkh, thanks for reporting it. I'll try to repro and get back soon.

**Comment by Nessex** [comment]
This also affects: `1.23.14-gke.401`

In addition, the pause image specified by GCP includes both a tag and a digest, which cri-o doesn't like:
https://github.com/cri-o/cri-o/issues/2351

The problem appears to be logic for extracting kubelet args in `/opt/crio-deploy/scripts/kubelet-config-helper.sh`, which is not present in this repo, but is present in the built sysbox image.

I suspect it's because the `pod-infra-container-image` value is the last value in an ENV string that looks something like this:

```
KUBELET_OPTS="--foo=bar --pod-infra-container-image=gke.gcr.io/pause:3.6@sha256:10008c36b4611b44db1229451675d5d7d86c7ddf4ef00f883d806a01547203f6"
```

Because it's the end of the string, there's an extra quote at the end. But the logic for extracting args doesn't account for that.

We put some super hacky workarounds in `kubelet-config-helper.sh`, like stripping the last quote from the pause_image if there are two, and stripping tags from the pause image if both a tag and digest are specified. I'm not sure where to contribute these as that is not a public script, but they probably need some careful attention from the sysbox team anyway.

For example:

 - Strip the tag from the pause image: `pause_image="$(echo "${pause_image_systemd}" | sed 's/:.*@/@/')"`
 - Strip the trailing quote on the pause image: `sed -i "s@pause_image =.*@pause_image = \"${pause_image%\"}\"@" $crio_conf_file`
 
The latter one in particular needs some work to account for cases where you might actually want that quote at the end of the value (e.g. when it's matched with a starting quote). A more elegant solution is probably necessary to support the general case, but specifically for the current GKE patch versions this should suffice.

**Comment by rodnymolina** [comment]
@Nessex, @joncbenderkh, could you please test this image to confirm that the two issues previously reported are fixed now?

You'll need to point our installation/uninstallation [manifests](https://github.com/nestybox/sysbox/tree/master/sysbox-k8s-manifests) to this image: `ghcr.io/nestybox/sysbox-deploy-k8s:rodny-dev`.

Thanks.

**Comment by joncbenderkh** [comment]
This does appear to have resolved the issue.  I tested on a few different  1.23.14 versions, as well as 1.25 and the deploys worked fine.

**Comment by rodnymolina** [comment]
Excellent @joncbenderkh, thank you.

**Comment by jamonation** [comment]
Just a drive-by +1 here - GKE repaved a v1.23.14-gke.1800 cluster and I ran into this. Running your dev image works well @rodnymolina, but I'm obviously a little wary of sticking with it for too long.

**Comment by rodnymolina** [comment]
Thanks for testing this out @jamonation. And yes, we are planning to release an official image within the next few days.

**Comment by RamyAllam** [comment]
Any ETA for the release?

**Comment by ctalledo** [comment]
Working on it right now; without mayor problems we should have it by next week.

**Comment by rodnymolina** [comment]
This issue has been fixed in the latest v0.6.1 image. Closing now.
